### PR TITLE
force cluster destroy -  vsphere platform

### PR DESCRIPTION
**Pull request by vavuthu**
This PR contains below important commits

5b72c33cc3e73c56520cbb6b58b957b442759f41 : cleanup tool for vsphere ( reading config file and establishes connection to vcenter )
0102875ea51053f3210cf4fa57330d7470984227 : Delete the disks from VM and delete the cluster from datacenter
fd7bf909df69c2a063537b72b716765afeb06288 : Release IP's from IPAM server
71d36ed475713b87549558edbba1e5a4d1bea288 : Delete hosted zone and its records from AWS
e2f7b4509173d6420ad875ad3e40263d3c122125 : Delete records from base domain

### CODE REVIEW

**Comment by vavuthu** [comment]
Verified locally

```
(venv1) [vavuthu@localhost ocs-ci](cluster_destroy_force_vsphere)$ vsphere-cleanup --cluster_name sshreeka-oct12 --vsphere_conf ~/VJ//VMWARE/UPLOAD/new/new/vSphere-DC-CP_VC1.yaml 

2020-10-20 11:41:41,741 - MainThread - ocs_ci.cleanup.vsphere.cleanup - INFO - Deleting disks from VM's
2020-10-20 11:42:15,027 - MainThread - ocs_ci.utility.vsphere - INFO - Detaching Disk with identifier: 1 from sshreeka-oct12-5m6jx-rhel-2 and remove from datastore=True
2020-10-20 11:42:26,584 - MainThread - ocs_ci.utility.vsphere - INFO - Detaching Disk with identifier: 2 from sshreeka-oct12-5m6jx-rhel-2 and remove from datastore=True
2020-10-20 11:42:42,384 - MainThread - ocs_ci.utility.vsphere - INFO - Detaching Disk with identifier: 1 from sshreeka-oct12-5m6jx-rhel-1 and remove from datastore=True
2020-10-20 11:42:50,698 - MainThread - ocs_ci.utility.vsphere - INFO - Detaching Disk with identifier: 2 from sshreeka-oct12-5m6jx-rhel-1 and remove from datastore=True
2020-10-20 11:43:35,370 - MainThread - ocs_ci.utility.vsphere - INFO - VM's in resource pool sshreeka-oct12: ['sshreeka-oct12-5m6jx-rhel-0', 'control-plane-2', 'control-plane-0', 'control-plane-1', 'compute-0', 'sshreeka-oct12-5m6jx-rhel-2', 'compute-1', 'compute-2', 'sshreeka-oct12-5m6jx-rhel-1', 'lb-0']
2020-10-20 11:43:36,987 - MainThread - ocs_ci.utility.vsphere - INFO - power state of sshreeka-oct12-5m6jx-rhel-0: poweredOn
2020-10-20 11:43:38,763 - MainThread - ocs_ci.utility.vsphere - INFO - power state of control-plane-2: poweredOn
2020-10-20 11:43:40,492 - MainThread - ocs_ci.utility.vsphere - INFO - power state of control-plane-0: poweredOn
2020-10-20 11:43:42,508 - MainThread - ocs_ci.utility.vsphere - INFO - power state of control-plane-1: poweredOn
2020-10-20 11:43:44,300 - MainThread - ocs_ci.utility.vsphere - INFO - power state of compute-0: poweredOn
2020-10-20 11:43:46,146 - MainThread - ocs_ci.utility.vsphere - INFO - power state of sshreeka-oct12-5m6jx-rhel-2: poweredOn
2020-10-20 11:43:47,923 - MainThread - ocs_ci.utility.vsphere - INFO - power state of compute-1: poweredOn
2020-10-20 11:43:49,662 - MainThread - ocs_ci.utility.vsphere - INFO - power state of compute-2: poweredOn
2020-10-20 11:43:51,257 - MainThread - ocs_ci.utility.vsphere - INFO - power state of sshreeka-oct12-5m6jx-rhel-1: poweredOn
2020-10-20 11:43:52,948 - MainThread - ocs_ci.utility.vsphere - INFO - power state of lb-0: poweredOn
2020-10-20 11:44:01,742 - MainThread - ocs_ci.utility.vsphere - INFO - Powering off VMs: ['sshreeka-oct12-5m6jx-rhel-0', 'control-plane-2', 'control-plane-0', 'control-plane-1', 'compute-0', 'sshreeka-oct12-5m6jx-rhel-2', 'compute-1', 'compute-2', 'sshreeka-oct12-5m6jx-rhel-1', 'lb-0']
2020-10-20 11:44:33,989 - MainThread - ocs_ci.utility.vsphere - INFO - Destroying VM's: ['sshreeka-oct12-5m6jx-rhel-0', 'control-plane-2', 'control-plane-0', 'control-plane-1', 'compute-0', 'sshreeka-oct12-5m6jx-rhel-2', 'compute-1', 'compute-2', 'sshreeka-oct12-5m6jx-rhel-1', 'lb-0']
2020-10-20 11:45:27,905 - MainThread - ocs_ci.utility.vsphere - INFO - Successfully deleted resource pool sshreeka-oct12
2020-10-20 11:45:27,905 - MainThread - ocs_ci.cleanup.vsphere.cleanup - INFO - Removing IP's for nodes ['sshreeka-oct12-5m6jx-rhel-0.sshreeka-oct12.qe.rh-ocs.com', 'control-plane-2.sshreeka-oct12.qe.rh-ocs.com', 'control-plane-0.sshreeka-oct12.qe.rh-ocs.com', 'control-plane-1.sshreeka-oct12.qe.rh-ocs.com', 'compute-0.sshreeka-oct12.qe.rh-ocs.com', 'sshreeka-oct12-5m6jx-rhel-2.sshreeka-oct12.qe.rh-ocs.com', 'compute-1.sshreeka-oct12.qe.rh-ocs.com', 'compute-2.sshreeka-oct12.qe.rh-ocs.com', 'sshreeka-oct12-5m6jx-rhel-1.sshreeka-oct12.qe.rh-ocs.com', 'lb-0.sshreeka-oct12.qe.rh-ocs.com'] from IPAM server
2020-10-20 11:45:27,907 - MainThread - urllib3.connectionpool - DEBUG - Starting new HTTP connection (1): 10.70.53.15:80
.
.
.
2020-10-20 11:45:43,927 - MainThread - /home/vavuthu/VJ/Gluster/automation/commits/MASTER1/ocs-ci/ocs_ci/utility/aws.py - INFO - Deleting record sshreeka-oct12.qe.rh-ocs.com. from qe.rh-ocs.com
2020-10-20 11:45:43,928 - MainThread - /home/vavuthu/VJ/Gluster/automation/commits/MASTER1/ocs-ci/ocs_ci/utility/aws.py - INFO - deleting record set: sshreeka-oct12.qe.rh-ocs.com.
2020-10-20 11:45:43,929 - MainThread - botocore.hooks - DEBUG - Event before-parameter-build.route-53.ChangeResourceRecordSets: calling handler <function fix_route53_ids at 0x7f1318af4b90>
2020-10-20 11:45:43,929 - MainThread - botocore.handlers - DEBUG - HostedZoneId /hostedzone/ZQ6XFE6BKI2L -> ZQ6XFE6BKI2L
2020-10-20 11:45:44,380 - MainThread - botocore.parsers - DEBUG - Response body:
b'<?xml version="1.0"?>\n<ChangeResourceRecordSetsResponse xmlns="https://route53.amazonaws.com/doc/2013-04-01/"><ChangeInfo><Id>/change/C0310878RG6JKNDHCW3M</Id><Status>PENDING</Status><SubmittedAt>2020-10-20T06:15:44.232Z</SubmittedAt></ChangeInfo></ChangeResourceRecordSetsResponse>'
2020-10-20 11:45:44,381 - MainThread - botocore.hooks - DEBUG - Event needs-retry.route-53.ChangeResourceRecordSets: calling handler <botocore.retryhandler.RetryHandler object at 0x7f130104d1d0>
2020-10-20 11:45:44,382 - MainThread - botocore.retryhandler - DEBUG - No retry needed.
(venv1) [vavuthu@localhost ocs-ci](cluster_destroy_force_vsphere)$ 
```

File: ocs_ci/cleanup/vsphere/cleanup.py [hunk 1]
```diff
@@ -0,0 +1,215 @@
+"""
```
Comment (ebenahar) [inline ocs_ci/cleanup/vsphere/cleanup.py:1]: thinking maybe we can create a class for each platform using factory design, the same as we have for deployment

**Review by ebenahar — APPROVED** [review]

**Comment by vasukulkarni** [comment]
conflict and needs rebase

**Comment by vavuthu** [comment]
> conflict and needs rebase

rebased to latest master

File: ocs_ci/cleanup/vsphere/cleanup.py (hunk 1 above)
Comment (vavuthu) [reply ocs_ci/cleanup/vsphere/cleanup.py:1]: I will take up this later as a separate PR

**Review by ebenahar — APPROVED** [review]

**Review by vasukulkarni — APPROVED** [review]

**Pull request merged**